metricbeat/docs/modules/system.asciidoc

```diff
@@ -177,6 +177,9 @@ metricbeat.modules:
   period: 10s
   processes: ['.*']
 
+  # Configure the mount point of the host’s filesystem for use in monitoring a host from within a container
+  #system.hostfs: "/hostfs"
```

Comment:
Have you considered adding this option to all beats as a global setting?

Apart of the system module, other features can benefit of accessing the host fs, as `add_docker_metadata`, that actually has [another setting](https://github.com/elastic/beats/blob/298886fa50512adb9392917e93711ae5417bafc5/libbeat/processors/add_docker_metadata/config.go#L37) for this, or the internal metrics, that [rely on the global `gosigar.Procd`](https://github.com/elastic/beats/blob/298886fa50512adb9392917e93711ae5417bafc5/libbeat/metric/system/process/process_linux.go#L30) to get the pid of the process. [`gosigar.Procd` is modified in Metricbeat by the system module](https://github.com/elastic/beats/blob/298886fa50512adb9392917e93711ae5417bafc5/metricbeat/module/system/system_linux.go#L43), what can lead to inconsistent behaviour between Beats (Metricbeat managed by an agent in a container can be monitored from the host pid namespace, but other Beats cannot).

An option can be to have explicit settings for each feature (as there is one for the system module and another one for `add_docker_metadata`) but I wonder if when wanting to override this setting this is not wanted for all features.

In any case, just food for thought, this could be decided later.

Reply:
Thanks for commenting @jsoriano ! Your suggestion sounds reasonable and definitely this configuration option could be revisited/refactored. 

For now I would avoid adding more complexity on this since this issue is quite blocking for running system module through Agent in containerized environements.

@fearful-symmetry do you think we should create a follow-up issue to file what Jaime suggested?